hub-tools/programmer/hub-programmer.c

```diff
@@ -115,39 +135,44 @@ int main (int argc, char*argv[])
     uint16_t hwid, hwid_dev, hwrev, hwrev_dev, fwver, fwver_dev;
     int iscrosslink = 0;
     
-    if (argc < 3)
+    if (argc < 5 || argc > 6)
     {
-        printf("Usage: %s <host_id> <hub_port> [binary_image]\n",argv[0]);
+        printf("Usage: %s <driver> <slot> <hub index> <binary_image> [-y]\n", argv[0]);
         exit(-1);
     }
 
-    hostid = atoi(argv[1]);
+    if (argc == 6)
+    {
+        ask_permission = strcmp(argv[5], "-y");
+    }
+
+    driver = argv[1];
+    slot = atoi(argv[2]);
 
-    port = atoi(argv[2]);
-    dev_idx = ((int)((port + 1) & 0x000000FF) << 8) + 254;
+    hub = atoi(argv[3]);
+    dev_idx = (hub << 8) + 254;
     if (argc < 4)
```

Comment:
Originally the bitfile was optional, and without it the program would print information about the hub. Do we want this feature removed? Since now the bitfile is mandatory. If it's the case, we should remove all logic related to this `noprog` option.

Reply:
Oops, I readded this ability